File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/SubstanceProtein.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class SubstanceProtein : DomainResource, IResource
{
    public SubstanceProtein_Subunit[]? Subunit { get; set; }
    public Base.CodeableConcept? SequenceType { get; set; }
    public string[]? DisulfideLinkage { get; set; }
    public int? NumberOfSubunits { get; set; }

    public class SubstanceProtein_Subunit : BackboneElement
    {
        public int? Length { get; set; }
        public int? Subunit { get; set; }
        public string? Sequence { get; set; }
        public Base.Attachment? SequenceAttachment { get; set; }
        public string? CTerminalModification { get; set; }
        public string? NTerminalModification { get; set; }
        public Base.Identifier? CTerminalModificationId { get; set; }
        public Base.Identifier? NTerminalModificationId { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/SubstanceProtein.py
from typing import Optional, List
from base import Attachment
from base import CodeableConcept
from base import DomainResource
from base import Identifier
from base import BackboneElement

class SubstanceProtein_Subunit(BackboneElement):
    length: Optional[int] = None
    subunit: Optional[int] = None
    sequence: Optional[str] = None
    sequence_attachment: Optional[Attachment] = None
    c_terminal_modification: Optional[str] = None
    n_terminal_modification: Optional[str] = None
    c_terminal_modification_id: Optional[Identifier] = None
    n_terminal_modification_id: Optional[Identifier] = None

class SubstanceProtein(DomainResource):
    subunit: Optional[List[SubstanceProtein_Subunit]] = None
    sequence_type: Optional[CodeableConcept] = None
    disulfide_linkage: Optional[List[str]] = None
    number_of_subunits: Optional[int] = None